src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using Xunit;
+// ReSharper disable InconsistentNaming
+
+namespace Tests.Collections
+{
+    public enum CollectionOrder
+    {
+        Unspecified,
+        Sequential
+    }
+
+    public static class CollectionAssert
+    {
+        public static void Equal(IEnumerable expected, IEnumerable actual)
+        {
+            Assert.Equal(expected.OfType<object>(), actual.OfType<object>());
+        }
+
+        public static void Equal(IList expected, IList actual)
+        {
+            Equal((IEnumerable)expected, actual);
+            Assert.Equal(expected.Count, actual.Count);
+            for (int i = 0; i < expected.Count; i++)
```

Comment:
I don't follow what this for loop is doing.

Isn't what it does already covered by the IEnumerable equals?

If not, it could probably use some commenting.